lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystem.java

```diff
@@ -322,6 +349,41 @@ public Optional<UriLocation> preSignedUri(Location location, Duration ttl)
         }
     }
 
+    @Override
+    public Optional<UriLocation> encryptedPreSignedUri(Location location, Duration ttl, EncryptionKey key)
```

Comment:
Make the above method delegate to this one, as it seems the only difference is setting the key